include/SDL3/SDL_gpu.h

```diff
@@ -941,212 +941,382 @@ typedef enum SDL_GPUDriver
+    float max_anisotropy;                      /**< The anisotropy value clamp used by the sampler. If enable_anisotropy is SDL_FALSE, this is ignored. */
+    SDL_bool enable_anisotropy;                /**< SDL_TRUE to enable anisotropic filtering. */
+    SDL_bool enable_compare;                   /**< SDL_TRUE to enable comparison against a reference value during lookups. */
     Uint8 padding1;
     Uint8 padding2;
-    SDL_GPUCompareOp compare_op;
-    float min_lod;
-    float max_lod;
+    SDL_GPUCompareOp compare_op;               /**< The comparison operator to apply to fetched data before filtering. */
+    float min_lod;                             /**< Clamps the minimum of the computed LOD value. */
+    float max_lod;                             /**< Clamps the maximum of the computed LOD value. */
 
-    SDL_PropertiesID props;
+    SDL_PropertiesID props;                    /**< A properties ID for extensions. Should be 0 if no extensions are needed. */
 } SDL_GPUSamplerCreateInfo;
 
+/**
+ * A structure specifying a vertex binding.
+ *
+ * When you call SDL_BindGPUVertexBuffers, you specify the binding indices of the vertex buffers.
+ * For example if you called SDL_BindGPUVertexBuffers with a first_binding of 2 and num_bindings of 3, the binding indices 2, 3, 4 would be used by the vertex buffers you pass in.
+ *
+ * Vertex attributes are linked to bindings via the index. The bindingIndex field of SDL_GPUVertexAttribute specifies the vertex buffer binding index that the attribute will be read from.
+ *
+ * \since This struct is available since SDL 3.0.0
+ *
+ * \sa SDL_GPUVertexAttribute
+ * \sa SDL_GPUVertexInputState
+ */
 typedef struct SDL_GPUVertexBinding
 {
-    Uint32 binding;
-    Uint32 pitch;
-    SDL_GPUVertexInputRate input_rate;
-    Uint32 instance_step_rate; /* ignored unless input_rate is INSTANCE */
+    Uint32 index;                     /**< The binding index. */
+    Uint32 pitch;                       /**< The byte pitch between consecutive elements of the vertex buffer. */
+    SDL_GPUVertexInputRate input_rate;  /**< Whether attribute addressing is a function of the vertex index or instance index. */
+    Uint32 instance_step_rate;          /**< The number of instances to draw using the same per-instance data before advancing in the instance buffer by one element. Ignored unless input_rate is SDL_GPU_VERTEXINPUTRATE_INSTANCE */
 } SDL_GPUVertexBinding;
 
+/**
+ * A structure specifying a vertex attribute.
+ *
+ * \since This struct is available since SDL 3.0.0
+ *
+ * \sa SDL_GPUVertexBinding
+ * \sa SDL_GPUVertexInputState
+ */
 typedef struct SDL_GPUVertexAttribute
 {
-    Uint32 location;
-    Uint32 binding;
-    SDL_GPUVertexElementFormat format;
-    Uint32 offset;
+    Uint32 location;                    /**< The shader input location index. */
+    Uint32 bindingIndex;                /**< The binding index. */
```

Comment:
```suggestion
    Uint32 binding_index;               /**< The binding index. */
```

Reply:
Old habits die hard...

Reply:
Yup. :)
If I were to write the API today, I'd use mixed case instead of underscores, but we tried to minimize structure field name changes in the transition to SDL3.

Reply:
Actually...